### PR TITLE
resolves #25: add more UDP tests

### LINKED ISSUE #25: Tests Connected UDP Sockets
1. If Sock-0 is connected to sock-1 try sending something (continuously) from Sock-2 to Sock-0 and confirm it's ignored and then after a while while Sock-2 is continuing to send to Sock-0 send something continuously from Sock-1 to Sock-0 and those should pass. This should give more confidence about cross-platform compliance of the behaviour should the test pass. Make sure you assert no errors are generated for Sock-0; it should simply filter the datagrams. Sock-1 should do `send()` and Sock-2 should do `send_to()` ofc.
1. Next (in a different test-case), Sock-0 should be able to `send()` and Sock-1 rx it. Simultaneously Sock-0 should be able to `send_to()` to Sock-2 and Sock-2 should rx it too. So while sending a connected socket (Sock-0) should be able to send to anyone but while rx only receive it from Sock-1. This behaviour might fail as I think connected Sock-0 will be disallowed to send to Sock-2 (because it cannot rx from it so OS stack might prevent sending to someone you can't rx from).

### PR DESCRIPTION
**Pull request by povilasb**

### CODE REVIEW

File: tests/udp.rs [hunk 1]
```diff
@@ -26,6 +27,124 @@ fn udp_peers_not_connected_huge_data_exchange() {
...
+    const SOCKET3_TOKEN: Token = Token(2);
+    const TIMEOUT_TOKEN: Token = Token(3);
+
+    let mut sock2 = unwrap!(UdpSock::bind(&unwrap!("127.0.0.1:0".parse())));
+    let sock2_addr = unwrap!(sock2.local_addr());
+
+    let mut sock1 = unwrap!(UdpSock::bind(&unwrap!("127.0.0.1:0".parse())));
+    unwrap!(sock1.connect(&sock2_addr));
+    let server_addr = unwrap!(sock1.local_addr());
+    unwrap!(sock2.connect(&server_addr));
```
Comment (ustulation) [inline tests/udp.rs]: is this necessary ? What is it trying to prove ?

File: tests/udp.rs [hunk 2]
```diff
@@ -26,6 +27,124 @@ fn udp_peers_not_connected_huge_data_exchange() {
...
+    'event_loop: loop {
+        let _ = unwrap!(poll.poll(&mut events, None));
+        for event in events.iter() {
+            match event.token() {
+                SOCKET1_TOKEN => loop {
+                    let res: Option<(Vec<u8>, _)> = unwrap!(sock1.read_frm());
+                    if let Some((_data, peer_addr)) = res {
+                        received_from.insert(peer_addr);
+                    } else {
+                        break;
```
Comment (ustulation) [inline tests/udp.rs]: why break out of the loop ? If it was a spurious wake and resulted in a WB error we want to continue, not break.

Comment (povilasb) [reply tests/udp.rs]: On the contrary, if it was `WouldBlock`, then we want to wait when the socket becomes readable again :)

Comment (povilasb) [reply tests/udp.rs]: meaning we call `read_frm()` until it returns `None`

File: tests/udp.rs (hunk 1 above)
Comment (povilasb) [reply tests/udp.rs]: on line 130 we're using `write()` instead of `write_to()` - in such case the socket must be connected

Comment (povilasb) [reply tests/udp.rs]: it's not necessary, it just exercises both APIs: write and write_to

File: tests/udp.rs (hunk 2 above)
Comment (ustulation) [reply tests/udp.rs]: Ya, what i meant was why break and not read the rest of the events in the event-set ? Other sockets could have their readiness triggered too.

Comment (povilasb) [reply tests/udp.rs]: it only break the inner loop on line 121 :)

Comment (ustulation) [reply tests/udp.rs]: Ahhhhh there's one more loop - i missed that.

File: tests/udp.rs (hunk 1 above)
Comment (ustulation) [reply tests/udp.rs]: Na let's keep the test specific in this case. Don't want it to convey wrong ms that both sided connections are somehow necessary. If you want you can do a specific test to test and contrast `write()/write_to()` but let's keep this one showing only the rx side's behaviour that the issue/test-function-name asks for.

**Pull request merged**